### PR TITLE
inline.d: move publics to top

### PR DESCRIPTION
**Pull request by WalterBright**
Make everything `private` we can, and move all `public` declarations to the top.

Just a copy/paste job.

### CODE REVIEW

File: src/dmd/inline.d [hunk 1]
```diff
@@ -40,8 +40,79 @@ import dmd.tokens;
...
+    }
+    scope ids = new InlineDoState(sc.parent, null);
+    return doInlineAs!Expression(e, ids);
+}
+
+
+
+
+
+
```
Comment (thewilsonator) [inline src/dmd/inline.d]: whitespace

**Review by thewilsonator — APPROVED** [review]

**Comment by Geod24** [comment]
The separation through whitespace is ugly and annoying. Really. No other module have that. And it's a good example about the pain of contributing to D: some day you decide to put a precedent in a module and suddenly some people start doing, some don't, and we have more ambiguities and inconsistencies. When other people try to change things at this level (e.g. @wilzbach ), they submit a PR affecting ALL the code that should be affected. It shows the change at scale, on a global level, not a macro one, and tells us if it really looks good. It also avoids confusing contributors on what should be done.

**Comment by Geod24** [comment]
Another (less-ranty) point: this adds `private` in front of functions which are already following a `private:` (at line 112). So the individual `private` are useless. It also adds `public` in front of public symbol, which is the default, and also rarely done elsewhere.

This is a more explicit style, which I personally like/favor in my projects, but again, diverging from what we usually do.

File: src/dmd/inline.d [hunk 2]
```diff
@@ -56,7 +127,7 @@ enum EXPANDINLINE_LOG = false;
  *  The best would be to return a pair of result Expression and a bool value as foundReturn
  *  from doInlineAs function.
  */
-final class InlineDoState
+private final class InlineDoState
```
Comment (jacob-carlborg) [inline src/dmd/inline.d]: All these symbols are already `private` due to `private:` on line 112.

File: src/dmd/inline.d [hunk 3]
```diff
@@ -40,8 +40,79 @@ import dmd.tokens;
...
 import dmd.inlinecost;
 
+/***********************************************************
+ * Scan function implementations in Module m looking for functions that can be inlined,
+ * and inline them in situ.
+ *
+ * Params:
+ *    m = module to scan
+ */
+public void inlineScanModule(Module m)
```
Comment (jacob-carlborg) [inline src/dmd/inline.d]: `public` it's not necessary since it's the default.

**Comment by Geod24** [comment]
@WalterBright : 

> Having the public interface come first, followed by the implementation details, is commonly considered good practice, as it makes it easy for someone browsing the code to see what the point of the module is.

You are completely missing the point I was making.
`public` first is good.
Weird precedents being introduced are not.
`public` first is not concerned, it has precedent and is widely accepted.
Random whitespace separation, and unnecessary `public` attribute has no or little precedent, and might be contentious.

**Comment by WalterBright** [comment]
> Weird precedents being introduced are not.

I have a hard time seeing a few blank lines as weird, painful, or contentious. I often use multiple blank lines to visually separate things, rather than use horizonal lines.

> unnecessary public attribute has no or little precedent, and might be contentious.

It's already used sporadically in the dmd source. Just grep `public` to see. I don't see how anyone would be confused by it. Especially since some files start out with `private:` at the top and then set certain functions to public, and most do not. I.e. there isn't a consistent practice.

**Pull request merged**